### PR TITLE
feat: Fix nightly builds and add to CI.

### PR DESCRIPTION
**Pull request by jsantell**
Preparing with in-flight PR https://github.com/getditto/safer_ffi/pull/191.

~~There's a new clippy lint rule (documented as available since 1.72.0, though only appears in nightly channels), and some extra finesse on targeting nightly builds with a dynamic lint.~~ also fixed in latest safer_ffi

Other than that, just a lifetime change needed.

### CODE REVIEW

File: rust/noosphere/build.rs [hunk 1]
```diff
@@ -1,6 +1,26 @@
 use cfg_aliases::cfg_aliases;
+use rustc_version::{version_meta, Channel};
 
 fn main() {
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rerun-if-env-changed=RUSTC");
+
+    // Enables `#[cfg(channel = "nightly")]` configuration.
+    match version_meta().unwrap().channel {
```
Comment (jsantell) [inline rust/noosphere/build.rs:9]: Maybe should just be `#[cfg(nightly)]` and `#[cfg(not(nightly))]`

**Comment by jsantell** [comment]
Upstream https://github.com/getditto/safer_ffi/pull/191 landed in [safer_ffi@0.1.4](https://github.com/getditto/safer_ffi/releases/tag/v0.1.4), moving from draft PR

**Comment by jsantell** [comment]
rebased on #685, updated with published safer-ffi 0.1.4 containing nightly fix, as well as the removal of a lint rule skip (also fixed in 0.1.4)

**Review by cdata — APPROVED** [review]
LGTM

**Pull request merged**